src/component/TurnipCalculator.tsx

```diff
@@ -24,8 +29,18 @@ export default function TurnipCalculator(props: any) {
   `;
 
   useEffect(() => {
-    setBuyPrice(price * turnips);
-  }, [price, turnips]);
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      setBuyPrice(price * turnips);
+    } else {
+      setAlertPrice(price === 0);
+      setAlertTurnip(turnips === 0);
+
+      if (!alertPrice && !alertTurnip) {
```

Comment:
This `setBuyPrice` ends up being fired anyways, and I believe it's because whenever we update state we are rerendering the whole thing, so when price gets set to 0 for example, this effect runs and updates the buy price since the alert booleans haven't been updated at that point.

I'm not really sure if this check should be here anyways because if either one is 0, the buy price will be 0 anyways and the calculation result (and the tip display) will not appear. What do you think?

Reply:
Not sure if understood you...I think you want to say is that the same `useEffect` 
```
  useEffect(() => {
    if (firstUpdate.current) {
      firstUpdate.current = false;
      setBuyPrice(price * turnips);
    } else {
      setAlertPrice(price === 0);
      setAlertTurnip(turnips === 0);

      if (!alertPrice && !alertTurnip) {
        setBuyPrice(price * turnips);
      }
    }
  }, [price, turnips, alertPrice, alertTurnip]);
```
runs twice, one for the price and another one for the price/turnips alert. I think is better leave it that way (for now) in order to ensure that the the correct execution order. If we have two different `useEffect` (one for the error alerts and another one for the price) we could get troubles of execution order. What do you think?

Reply:
Yeah sorry, I definitely wrote a confusing comment haha, even confused myself looking at it now. :P but what I think I was trying to say was that every time price/turnip is updated from zero to nonzero or vice-versa, the `else` block in the `useEffect` will run twice, once after the price/turnip is updated and the second time after the corresponding alert boolean is changed. I think it's fine though, thinking about it now, because the buy price will still be updated, and won't be updated twice.

I agree, let's keep it. 👍